### ISSUE TITLE
bug: ion-input should inherit aria-label to inner input

### ISSUE THREAD

**Issue by meomar** [ionic-team/ionic-framework#22814]
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Current behavior:**
There is no way to provide  aria-label to the input element of ion-input component.

**Expected behavior:**
<img width="973" alt="Screenshot 2021-01-25 at 12 47 08" src="https://user-images.githubusercontent.com/10757713/105696097-883e5f00-5f0b-11eb-966b-03437ca8f226.png">

```
<ion-input ariaLabel="example"></ion-input>
```

**Comment by liamdebeasi** [comment]
Thanks for the issue. Can you provide an example of how you are using `ion-input`? If you use it with an `ion-label`, screen readers will read the value of `ion-label`.

**Comment by meomar** [comment]
> Thanks for the issue. Can you provide an example of how you are using `ion-input`? If you use it with an `ion-label`, screen readers will read the value of `ion-label`.

Actually, you are correct - the screenreader is reading the label, but what made me thinking it is not reading, it is that I got violations (Invalid ARIA attribute value) from different tools like Axe. The only way to pass this error is when I have aria-label on the actual input element.

Example : 

```
<ion-item>
  <ion-label>Default Label</ion-label>
  <ion-input></ion-input>
</ion-item>

```

**Comment by meomar** [comment]
Seems the same issue as : https://github.com/ionic-team/ionic-framework/issues/21569

**Comment by liamdebeasi** [comment]
Thanks! Do you have the full html of the page that you are seeing this error? I can't seem to get the "Invalid ARIA attribute value" error when I test the code snippet with Axe.

**Comment by meomar** [comment]
> Thanks! Do you have the full html of the page that you are seeing this error? I can't seem to get the "Invalid ARIA attribute value" error when I test the code snippet with Axe.

Okay so... it is only reproduced when you wrap ion-label and ion-input using regular div instead of ion-item:

	<div class="filter">
		<ion-label>Default Label</ion-label>
		<ion-input></ion-input>
	</div>

**Comment by liamdebeasi** [comment]
Yes it is required. Without an `ion-item` we are unable to properly associate the `ion-label` and `ion-input` together. If you were to use `ion-item`, you should notice that `ion-label` has an ID like "ion-label-0-lbl". Without the `ion-item`, the `ion-label` should not have that ID. This ID is what is added to the `aria-labelledby` attribute on the `input` inside of `ion-input`. 

Can you try your example again with an `ion-item` and let me know if it that resolves the issue?

**Comment by meomar** [comment]
> Yes it is required. Without an `ion-item` we are unable to properly associate the `ion-label` and `ion-input` together. If you were to use `ion-item`, you should notice that `ion-label` has an ID like "ion-label-0-lbl". Without the `ion-item`, the `ion-label` should not have that ID. This ID is what is added to the `aria-labelledby` attribute on the `input` inside of `ion-input`.
> 
> Can you try your example again with an `ion-item` and let me know if it that resolves the issue?

Yes thanks! it is solving this issue but what about the case when i want to show just ion-input with some placeholder and i don't want to show the label to the user.

`<ion-input placeholder="Enter First Name"></ion-input>`

in this case Axe is complying about: 

Invalid ARIA attribute value: aria-labelledby="ion-input-0-lbl" 

can we use ion-input separated from ion-label ?

**Comment by liamdebeasi** [comment]
That's a good point - I think we can improve how we handle custom aria labels. For `ion-button` we have `aria-label` get inherited to the inner native `button` element, so I think we can do the same thing here: https://github.com/ionic-team/ionic-framework/blob/818e387fe81ac7026fb374d8865116dadd433c87/core/src/components/button/button.tsx#L138

**Comment by meomar** [comment]
Sounds good, thanks.

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic-framework/pull/23159, and a fix will be available in an upcoming release of Ionic Framework.
